File: plugin/jvm/src/main/java/com/twosigma/beaker/chart/xychart/plotitem/Area.java
/*
 *  Copyright 2014 TWO SIGMA OPEN SOURCE, LLC
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package com.twosigma.beaker.chart.xychart.plotitem;

import com.twosigma.beaker.chart.Color;
import com.twosigma.beaker.chart.Filter;

import java.util.EnumSet;
import java.util.List;


public class Area extends XYGraphics {

  private final static EnumSet<Filter> POSSIBLE_LOD_FILTERS = EnumSet.of(Filter.AREA, Filter.RIVER);

  private Number baseBase = 0.0d;
  private List<Number> bases = null;
  private Integer interpolation;

  public void setBase(Object base) {
    if (base instanceof Number) {
      this.baseBase = (Number) base;
    } else if (base instanceof List) {
      @SuppressWarnings("unchecked")
      List<Number> temp = (List<Number>) base;
      setBases(temp);
    } else {
      throw new IllegalArgumentException(
          "setBase takes Number or List of Number");
    }
  }

  private void setBases(List<Number> bases) {
    this.bases = bases;
  }

  public Number getBase() {
    return this.baseBase;
  }

  public List<Number> getBases() {
    return this.bases;
  }

  public void setInterpolation(Integer interpolation) {
    if (interpolation.intValue() < 0 || interpolation.intValue() > 1) {
      throw new IllegalArgumentException(
          "Area interpolation is limited to 0, 1");
    }

    this.interpolation = interpolation;
  }

  public Integer getInterpolation() {
    return this.interpolation;
  }

  @Override
  protected EnumSet<Filter> getPossibleFilters() {
    return POSSIBLE_LOD_FILTERS;
  }
}

File: kernel/scala/src/main/java/com/twosigma/beakerx/scala/chart/xychart/plotitem/Area.scala
/*
 *  Copyright 2017 TWO SIGMA OPEN SOURCE, LLC
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *         http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 */

package com.twosigma.beakerx.scala.chart.xychart.plotitem

import com.twosigma.beakerx.chart.Color

import scala.collection.JavaConverters._

class Area extends com.twosigma.beakerx.chart.xychart.plotitem.Area {

  def this(x: List[Int], y: List[Int]) {
    this()
    super.setX(x.map(x => x.asInstanceOf[AnyRef]).asJava)
    super.setY(y.map(x => x.asInstanceOf[Number]).asJava)
  }

  def this(x: List[Int], y: List[Int], color: Color, interpolation: Int) {
    this(x, y)
    super.setColor(color)
    super.setInterpolation(interpolation)
  }

  def this(y: List[Int], base: List[Int], color: Color, displayName: String) {
    this()
    super.setY(y.map(x => x.asInstanceOf[Number]).asJava)
    super.setBase(base.map(x => x.asInstanceOf[AnyRef]).asJava)
    super.setColor(color)
    super.setDisplayName(displayName)
  }

  def this(y: List[Int], displayName: String) {
    this()
    super.setY(y.map(x => x.asInstanceOf[Number]).asJava)
    super.setDisplayName(displayName)
  }



}
